### DISCUSSION TITLE
Access basic blocks in javascript block

### DISCUSSION THREAD

**Discussion by ThaddaeusWiedemer** [Obsidian-TTRPG-Community/fantasy-statblocks#390] [Q&A] [answered]
I would like to reuse some computed values (like the proficiency bonus in DnD 5e that is derived from `monster.cr`) in multiple places throughout the statblock and define functions that I can use in multiple blocks. For this reason, I am tempted to simply define the entire statblock in a javascript block and use a custom .css.

However, many of the basic blocks are still useful and would save a lot of time in creating a statblock like this. Therefore, I'm wondering whether it is possible to somehow programmatically create some of the basic blocks within a javascript block. I am imagining something like:

```javascript
const myStatblock = document.createElement('div');
const spellBlock = fantasystatblocks.spellBlock(property=monster.spells; parseForDices=true);
myStatblock.appendChild(spellBlock);
return myStatblock;
```

Is this possible at all?

**Comment by valentine195** [comment, marked as answer]
This is a pretty good idea. Let me think through the structure of it.

In the mean time, you can use computed values throughout the layout by assigning them to the `monster` object in the JS callback. The `monster` object is a cloned version of the underlying data, but the same object is used throughout the layout. If your JS block is the first block, anything you set there will be available in all the other blocks.

**Reply by ThaddaeusWiedemer** [reply]
Thanks for the speedy reply, that's a really good workaround that I hadn't thought of!
